### PR TITLE
azure: migrated provider to newer client library, remove gocloud.dev dependency

### PR DESCRIPTION
**Pull request by jkowalski**

### CODE REVIEW

File: internal/gather/gather_bytes.go [hunk 1]
```diff
@@ -114,26 +114,57 @@ func (b Bytes) ReadAt(p []byte, off int64) (n int, err error) {
 	return len(p), b.AppendSectionTo(bytes.NewBuffer(p[:0]), int(off), len(p))
 }
 
-// Reader returns a reader for the data.
-func (b Bytes) Reader() io.Reader {
-	b.assertValid()
+type bytesReadSeekCloser struct {
```
Comment (julio-lopez) [inline internal/gather/gather_bytes.go:117]: `bytes.Reader` already implements `io.ReadSeeker`. It's only missing `Close()`. Why not extend `bytes.Reader` with just `Close()`? What am I missing?

**Review by julio-lopez — COMMENTED** [review]
+1: LG 👍

See question inline

CI failed due to Azure license. Needs to be checked.

File: internal/gather/gather_bytes.go (hunk 1 above)
Comment (jkowalski) [reply internal/gather/gather_bytes.go:117]: there's still a case of MultiReader, which don't implement io.ReadSeeker unfortunately.

File: .wwhrd.yml [hunk 2]
```diff
@@ -14,3 +14,6 @@ allowlist:
   - MPL-2.0
   - CC0-1.0
   - ISC
+
+exceptions:
+ - github.com/Azure/azure-sdk-for-go/sdk/internal/...
```
Comment (julio-lopez) [inline .wwhrd.yml:19]: nit: add EOL

**Pull request merged**